Dev normalize predicate (#804)
neo4j/docs-cypher: modules/ROOT/pages/clauses/where.adoc

|===

[[match-string-is-normalized]]

> === Checking if a `STRING` `IS NORMALIZED`

Reviewer: "`STRING`" has mostly been written as "string" in other headers

> The `IS NORMALIZED` operator is used to check whether the given `STRING` is in the `NFC` Unicode normalization form:

Reviewer: same comment about `STRING` - usually just written as "string" on this page

Reviewer 2: I will fix this in a separate PR (have carded it) - it should be `STRING` as per the new formatting style. This page in general is in a pretty dire need of restructuring, so will try to get to it sooner rather than later. I think @gem-neo4j PR can go in as is for now.

|===

[[match-string-is-normalized]]
=== Checking if a string is normalized `IS NORMALIZED`

The `IS NORMALIZED` operator is used to check whether the given `STRING` is in the `NFC` Unicode normalization form:

.Query
[source, cypher]
----
MATCH (n:Person)
WHERE n.name IS NORMALIZED
RETURN n.name AS normalizedNames
----

> The given strings contain only normalized Unicode characters, therefore all the names are matched and returned.

Reviewer 2: ```suggestion
The given `STRING` values contain only normalized Unicode characters, therefore all the matched `name` properties are returned.
```

> See here for more information on the xref:syntax/operators.adoc#match-string-is-normalized[normalization operator].

Reviewer 2: ```suggestion
For more information, see the section about the xref:syntax/operators.adoc#match-string-is-normalized[normalization operator].
```

.Result
[role="queryresult",options="header,footer",cols="1*<m"]
|===
| normalizedNames
| 'Andy'
| 'Timothy'
| 'Peter'
2+|Rows: 1
|===

> *Considerations:*
> 
> |===
> 
> | `IS NORMALIZED` returns `null` when used on a non-`STRING` value, e.g. `RETURN 1 IS NORMALIZED` returns `null`.
> 
> |===

Reviewer 2: ```suggestion
Note that the `IS NORMALIZED` operator returns `null` when used on a non-`STRING` value.
For example,  `RETURN 1 IS NORMALIZED` returns `null`.
```
